src/webapi/date.rs

```diff
@@ -1,22 +1,903 @@
+                return @{self.as_ref()}.setUTCMinutes(@{args[0]}, @{args[1]}, @{args[2]});
+            ).try_into().unwrap(),
+            2 => js!( return @{self.as_ref()}.setUTCMinutes(@{args[0]}, @{args[1]}); ).try_into().unwrap(),
+            1 => js!( return @{self.as_ref()}.setUTCMinutes(@{args[0]}); ).try_into().unwrap(),
+            _ => js!( return @{self.as_ref()}.setUTCMinutes(); ).try_into().unwrap()
+
+        }
+    }
+
+    ///
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/setUTCMonth)
+    #[inline]
+    fn set_utc_month( &self, args: &[i32] ) -> f64 {
+        // args represents:
+        // [<month_value>, <day_value>]
+        match args.len() {
+            2 => js!( return @{self.as_ref()}.setUTCMonth(@{args[0]}, @{args[1]}); ).try_into().unwrap(),
+            1 => js!( return @{self.as_ref()}.setUTCMonth(@{args[0]}); ).try_into().unwrap(),
+            _ => js!( return @{self.as_ref()}.setUTCMonth(); ).try_into().unwrap()
+
+        }
+    }
+
+    ///
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/setUTCSeconds)
+    #[inline]
+    fn set_utc_seconds( &self, args: &[i32] ) -> f64 {
+        // args represents:
+        // [<seconds_value>, <ms_value>]
+        match args.len() {
+            2 => js!( return @{self.as_ref()}.setUTCSeconds(@{args[0]}, @{args[1]}); ).try_into().unwrap(),
+            1 => js!( return @{self.as_ref()}.setUTCSeconds(@{args[0]}); ).try_into().unwrap(),
+            _ => js!( return @{self.as_ref()}.setUTCSeconds(); ).try_into().unwrap()
+
+        }
+    }
+
+    /// DEPRECATED
+    /// fn set_year() -> f64 { }
+
+    ///
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toDateString)
+    #[inline]
+    fn to_date_string( &self ) -> String {
+        js! ( return @{self.as_ref()}.toDateString; ).try_into().unwrap()
+    }
+
+    /// DEPRECATED
+    /// fn to_gmt_string( &self ) -> f64 { }
+
+    ///
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString)
+    #[inline]
+    fn to_iso_string( &self ) -> String {
+        js! ( return @{self.as_ref()}.toISOString(); ).try_into().unwrap()
+    }
+
+    ///
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toJSON)
+    #[inline]
+    fn to_json( &self ) -> String {
+        js! ( return @{self.as_ref()}.toJSON(); ).try_into().unwrap()
+    }
+
+    /// to_locale_date_string
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString)
+    #[inline]
+    fn to_locale_date_string( &self, _locales: Option< &str >, _options: Option< &HashMap< String, String > >) -> String {
```

Comment:
Uh, this function (on the JS side) looks super complex to reasonably bind in Rust. I definitely wouldn't want to have a stringly-typed prototype like this one.

My suggestion would be to just remove this one altogether for now. (If someone *really* wants to use it he/she can always use the `js!` macro.)

Reply:
I have mixed feelings on this one. Overall I don't mind building a typed hash map, but to your point I could see it getting repetitive/verbose. On the flip side, I like being able to utilize compile-time checks to make sure function calls/signatures make sense